### ISSUE TITLE
DVC 2.10.2 unpinned dependencies causes issues with file imports

### ISSUE THREAD

**Issue by ryanswrt** [NixOS/nixpkgs#175885]
### Describe the bug
The dvc package unpins scmrepo and grandalf; which will cause issues according to their developers. For dvc 2.10.2 ; scmrepo /has/ to be at  0.0.19 due to pathing changes

The particular issue I am experiencing is that dvc fails to import files from git repos in the current 22.05 / unstable nixpkgs

### Steps To Reproduce
1. `mkdir dvcbug && cd dvcbug && git init`
2. `nix run nixpkgs#dvc -- init`
3. `nix run nixpkgs#dvc -- import git@github.com:iterative/example-get-started data/data.xml`
```console
Importing 'data/data.xml (git@github.com:iterative/example-get-started)' -> 'data.xml'
ERROR: failed to import 'data/data.xml' from 'git@github.com:iterative/example-get-started'. - The path 'data/data.xml' does not exist in the target repository 'git@github.com:iterative/example-get-started' neither as a DVC output nor as a Git-tracked file.: [Errno 2] No such file or directory: '/tmp/tmpyzehuj22dvc-clone/data/data.xml'
```

### Expected behavior
File to be imported correctly

### Additional context
This used to work around `3ed7f12b5f926430015c4fb185b3288f6689c4c8`

### Notify maintainers
@CMCDragonkai @fab

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[machineer@nixos:~/Programming/dvcbug]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.06 MiB download, 0.30 MiB unpacked):
  /nix/store/yjgdgysbfm79s3a5xjawcvz2qyfgpiw6-bash-interactive-5.1-p16-dev
copying path '/nix/store/yjgdgysbfm79s3a5xjawcvz2qyfgpiw6-bash-interactive-5.1-p16-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.11, NixOS, 22.05 (Quokka), 22.05.342.a634c8f6c1f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(machineer): `"TheAncientGoat"`
 - channels(root): `"nixos-22.05, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by TheAncientGoat** [comment]
Context from conversation with devs https://discord.com/channels/485586884165107732/981479270008356864

**Comment by fabaff** [comment]
@TheAncientGoat, could you please check #175945. I added an override for `grandalf` and `scmrepo` to get the exact releases which `dvc` wants.

**Comment by TheAncientGoat** [comment]
Looks like it works; thanks - btw there's another issue - for the google cloud storage to work gcfs is needed as a dep like I have here https://github.com/TheAncientGoat/nixpkgs/commit/1f035bf9afd55493a2c4e46caab78131ccbfb7b2 ; shall I make an issue for that as well?

**Comment by fabaff** [comment]
> shall I make an issue for that as well?

No need. I adjusted the dependencies in https://github.com/NixOS/nixpkgs/pull/175945/commits/f169b0ed7891fd64510e5aab448c4029fa3f7443

Thanks for your feedback.

**Comment by fabaff** [comment]
Unfortunatly does the fix depend on #174098 now.

**Comment by TheAncientGoat** [comment]
Unsure of whether the fix depends on that; seems to affect a lot of packages

**Comment by fabaff** [comment]
The updates for the dependencies went to `python-updates` and will go into `staging-next`soon.

**Comment by fabaff** [comment]
I'm going to split the existing PR (https://github.com/NixOS/nixpkgs/pull/175945) into multiple PRs.

**Comment by fabaff** [comment]
Re-open because of the missing extra dependencies.
